src/blameandshame/observation.py

```diff
@@ -77,29 +80,25 @@ def modified_lines(self) -> FrozenSet[Line]:
                                                       in_files=files)
 
 
-class ObservationCollection(object):
-    @staticmethod
-    def load(fn: str) -> 'ObservationCollection':
-        observations = []
-        with open(fn, 'r') as f:
+class ObservationCollection(Generator):
+    def __init__(self, filename: str) -> None:
+        with open(filename, 'r') as f:
             reader = csv.reader(f)
             # skip header
             next(reader, None)
-            for row in reader:
-                # We aren't using the Travis URLs yet
-                repo_url, bug_sha, bug_build_url, fix_sha, fix_build_url = row
-                observation = Observation.build(repo_url, bug_sha, fix_sha)
-                observations.append(observation)
-        return ObservationCollection(observations)
-
-    def __init__(self, observations: List[Observation]) -> None:
-        self.__observations = observations.copy()
+            observations = list(reader)
+        self.__observations = iter(observations)
```

Comment:
We'll have to create a fresh observation collection every time? Seems error prone. Is that the intent, or is the intent to lazily create the Observation instances?